File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer,Integer> numMap = new HashMap<>();
        for(int num : nums){
           numMap.put(num, numMap.getOrDefault(num,1)+1);
        }
        PriorityQueue<Integer> heap = new PriorityQueue<>((a,b) -> numMap.get(a)-numMap.get(b));

        for (int num : numMap.keySet()) {
            heap.add(num);
            if (heap.size() > k) {
                heap.poll(); // Remove the element with the lowest frequency
            }
        }

        //To retain top k max elements we need min heap to delete elements when size exceeds k
        // Step 4: Extract the top k elements from the heap
        int[] result = new int[k];
        for (int i = 0; i < k; i++) {
            result[i] = heap.poll();
        }
        return result;
    }
}

File: Data Structures & Algorithms/house-robber/submission-1.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        size = len(nums)
        dp = [0] * size
        if size >= 2:
            dp[size-1] = nums[size-1]
            dp[size-2] = max(nums[size-2], nums[size-1])
        else:
            return sum(nums)

        for i in range(size-3, -1, -1):
            dp[i] = max(nums[i] + dp[i+2], dp[i+1])

        return dp[0]